service/debugger/debugger.go

```diff
@@ -1475,6 +1486,73 @@ func (d *Debugger) Functions(filter string) ([]string, error) {
 	return funcs, nil
 }
 
+func filterRuntimeFuncs(fname string) bool {
+	return !((strings.HasPrefix(fname, "runtime.") || strings.HasPrefix(fname, "runtime/internal")) && fname != "runtime.deferreturn" && fname != "runtime.gorecover" && fname != "runtime.gopanic")
+
+}
+
+func traverse(t proc.ValidTargets, f *proc.Function, depth int, followCalls int) ([]string, error) {
+	type TraceFunc struct {
+		Func    *proc.Function
+		Depth   int
+		visited bool
+	}
+	type TraceFuncptr *TraceFunc
+
+	TraceMap := make(map[string]TraceFuncptr)
+	queue := make([]TraceFuncptr, 0, 40)
+	funcs := []string{}
+	rootnode := &TraceFunc{Func: new(proc.Function), Depth: depth, visited: false}
+	rootnode.Func = f
+
+	// cache function details in a map for reuse
+	TraceMap[f.Name] = rootnode
+	queue = append(queue, rootnode)
+	for len(queue) > 0 {
+		parent := queue[0]
+		queue = queue[1:]
+		if parent == nil {
+			panic("attempting to open file Delve cannot parse")
+		}
+		if parent.Depth > followCalls {
+			continue
+		}
+		if !parent.visited {
+			funcs = append(funcs, parent.Func.Name)
+			parent.visited = true
+		} else if parent.visited {
+			continue
+		}
+
+		if parent.Depth+1 > followCalls {
+			// Avoid diassembling if we already cross the follow-calls depth
+			continue
+		}
+		f := parent.Func
+		text, err := proc.Disassemble(t.Memory(), nil, t.Breakpoints(), t.BinInfo(), f.Entry, f.End)
+		if err != nil {
+			return nil, fmt.Errorf("disassemble failed with error %w", err)
+		}
+		for _, instr := range text {
+			if instr.IsCall() && instr.DestLoc != nil && instr.DestLoc.Fn != nil {
+				cf := instr.DestLoc.Fn
+				if !filterRuntimeFuncs(cf.Name) {
+					continue
+				}
+				childnode := TraceMap[cf.Name]
+				if childnode == nil {
+					childnode = &TraceFunc{Func: new(proc.Function), Depth: parent.Depth + 1, visited: false}
```

Comment:
It doesn't make sense to initialize Func with `new(proc.Function)` only to replace it with `cf` in the very next line.